### PR TITLE
Capture extra metadata info for engine releases.

### PR DESCRIPTION
**Pull request by mgiannakopoulos**
Adding script for downloading `BOTTLE_VERSION` release file from `memsql` repo and uploading extra metadata info of engine releases (which include the maximum version of bottleAPI version to be used for communication between engine and bottle-service) to the `helios-infra-engine-releases-metadata` s3 bucket.

The extra steps for advertising the bottle-version dependency for engine release 8.5.x to control-plane have been added to the existing `publish-gcr-node` make target.

cc: @jvictor0 @Cynary @shiyuluu @FranciscoMSM @mojdehmemsql @adbirka @eli-memsql @nrajkumar1345 @ramrajprabu @acorreia-singlestore @klung16

JIRA Tasks: https://memsql.atlassian.net/browse/DB-65157, https://memsql.atlassian.net/browse/DB-65362

### CODE REVIEW

File: assets/release-metadata/release-metadata.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        bottleVersionMinorStr = config['DEFAULT']['BOTTLE_VERSION_MINOR']
+        bottleVersionPatchStr = config['DEFAULT']['BOTTLE_VERSION_PATCH']
+        if not bottleVersionMajorStr.isdigit() or not bottleVersionMinorStr.isdigit() or not bottleVersionPatchStr.isdigit():
+            raise Exception("BOTTLE_VERSION file contents are invalid: %s" % bottleVersion)
+        releaseMetadataContents = {
+            'bottleVersion': '%s.%s.%s' % (bottleVersionMajorStr, bottleVersionMinorStr, bottleVersionPatchStr),
+        }
+        releaseMetadataFile = "%s_%s_%s.json" % (bottleVersionMajorStr, bottleVersionMinorStr, bottleVersionPatchStr)
+        generate_release_metadata_file(releaseMetadataFile, releaseMetadataContents)
+        upload_release_metadata_file(releaseMetadataS3Bucket, releaseMetadataBucketRegion, releaseMetadataAWSAccessKeyID, releaseMetadataAWSAccessKeySecret, releaseMetadataFile, memsqlServerVersion)
```
Comment (FranciscoMSM) [inline assets/release-metadata/release-metadata.py:65]: add error handling around the aws operations

File: Makefile [hunk 2]
```diff
@@ -536,3 +536,11 @@ requires-license:
 ifndef LICENSE_KEY
 	$(error LICENSE_KEY is required)
 endif
+
+.PHONY: test-engine-extra-metadata
+test-engine-extra-metadata:
+	MEMSQL_SERVER_VERSION=${SERVER_VERSION_PREVIEW} \
+	RELEASE_CHANNEL=dev \
+	RELEASE_METADATA_BUCKET=helios-infra-engine-releases-metadata \
+	RELEASE_METADATA_BUCKET_REGION=us-east-1 \
```
Comment (FranciscoMSM) [inline Makefile:545]: bucket and region should come from env variables (configured in circle ci)

**Review by FranciscoMSM — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mgiannakopoulos) [reply Makefile:574]: It appears that apart from exposing those two env variables via CircleCI we will need also to expose the static secrets that allows us to access the S3 bucket in CircleCI. Currently jobs are failing with the following error:
```
MEMSQL_SERVER_VERSION=8.5.1-417dd55bd4 \
RELEASE_CHANNEL=dev \
RELEASE_METADATA_BUCKET=helios-infra-engine-releases-metadata \
RELEASE_METADATA_BUCKET_REGION=us-east-1 \
./assets/release-metadata/release-metadata.py
ERROR:root:Missing required environment variable RELEASE_METADATA_AWS_ACCESS_KEY_SECRET
Traceback (most recent call last):
  File "./assets/release-metadata/release-metadata.py", line 55, in <module>
    releaseMetadataAWSAccessKeySecret = _must_get_env("RELEASE_METADATA_AWS_ACCESS_KEY_SECRET", redact_log=True)
  File "./assets/release-metadata/release-metadata.py", line 15, in _must_get_env
    raise Exception("Environment variable {} is required".format(var))
Exception: Environment variable RELEASE_METADATA_AWS_ACCESS_KEY_SECRET is required

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./assets/release-metadata/release-metadata.py", line 72, in <module>
    if os.path.exists(releaseMetadataFile):
NameError: name 'releaseMetadataFile' is not defined
make: *** [Makefile:571: test-engine-extra-metadata] Error 1

Exited with code exit status 2
```

https://app.circleci.com/pipelines/github/memsql/deployment-docker/662/workflows/8de566dc-3140-4a60-b253-6f182b5f293a/jobs/3960

Comment (mgiannakopoulos) [reply Makefile]: Done!

File: assets/release-metadata/release-metadata.py (hunk 1 above)
Comment (mgiannakopoulos) [reply assets/release-metadata/release-metadata.py]: Done!

**Pull request merged**